improver/psychrometric_calculations/total_precipitable_water.py

```diff
@@ -0,0 +1,169 @@
+# (C) Crown Copyright, Met Office. All rights reserved.
+#
+# This file is part of 'IMPROVER' and is released under the BSD 3-Clause license.
+# See LICENSE in the root of the repository for full licensing details.
+"""Module to generate total precipitable water"""
+
+from typing import Tuple
+
+import iris
+import numpy as np
+from iris.cube import Cube
+from numpy import ndarray
+
+from improver import BasePlugin
+from improver.constants import EARTH_SURFACE_GRAVITY_ACCELERATION, WATER_DENSITY
+
+
+class PrecipitableWater(BasePlugin):
+    """
+    Plugin to calculate total precipitable water from a humidity mixing ratio cube.
+
+    This plugin integrates the humidity mixing ratio over pressure levels to compute
+    the total precipitable water (TPW) in metres. It assumes pressure levels are in
+    Pascals and uses constants for gravity and water density to convert the result
+    into liquid water equivalent thickness.
+
+    Attributes:
+        model_type (str): Type of model used, e.g., 'global' or 'ukv'.
+    """
+
+    def __init__(self, model_type: str = "global") -> None:
+        """
+        Initialise the PrecipitableWater plugin.
+
+        Args:
+            model_type:
+                Type of model used, e.g., 'global' or 'ukv'.
+        """
+        self.model_type = model_type.lower()
+
+    @staticmethod
+    def calculate_pressure_thickness(
+        layer_bounds: ndarray, cube_data: ndarray
+    ) -> ndarray:
+        """
+        Calculate the pressure thickness of each atmospheric layer.
+
+        Args:
+            layer_bounds:
+                1D array of pressure layer boundaries in Pascals.
+            cube_data:
+                N-dimensional array of humidity mixing ratio values.
+
+        Returns:
+            N-dimensional array of pressure thickness values reshaped to match
+            the dimensions of the input data.
+        """
+        delta_p = -np.diff(layer_bounds)
```

Comment:
This removes the need for ascending pressure coord points:
```suggestion
        delta_p = np.abs(np.diff(layer_bounds))
```